### PR TITLE
Add ruff rules and fix shape module and rve

### PR DESCRIPTION
**Pull request by kmarchais**
This PR adds all the ruff rules and fixes the issues spotted by these rules in the shape subpackage and the rve file.

The goal will be to remove files from the ignore parameter in the `[tool.ruff.lint]` section in `pyproject.toml` step by step.

### CODE REVIEW

File: pyproject.toml
Comment (chemiskyy) [inline pyproject.toml:1]: "microgen/phase.py" is excluded 2 times

File: tests/test_mesh_periodic.py [hunk 1]
```diff
@@ -92,11 +92,9 @@ def _generate_cqcompound_octettruss(rve: Rve):
 @pytest.fixture(scope="function")
 def box_homogeneous_unit(rve_unit: Rve) -> Tuple[cq.Shape, List[Phase], Rve]:
     shape = Box(
-        center=rve_unit.center,
+        center=(rve_unit.center[0], rve_unit.center[1], rve_unit.center[2]),
```
Comment (chemiskyy) [inline tests/test_mesh_periodic.py:95]: is there a possible typing issue there?

**Review by chemiskyy — APPROVED** [review]

File: tests/test_mesh_periodic.py (hunk 1 above)
Comment (kmarchais) [reply tests/test_mesh_periodic.py:95]: Fixed, thanks

File: microgen/shape/__init__.py [hunk 2]
```diff
@@ -16,47 +17,45 @@
...
+        float
+        | Sequence[float]
+        | tuple[float, float, float]
+        | Sequence[tuple[float, float]]
+        | Callable
+    )
+
 
-def newGeometry(
+def new_geometry(  # noqa: PLR0911
```
Comment (ricardo0115) [inline microgen/shape/__init__.py:52]: This function needs to be removed for release 2.0

**Review by ricardo0115 — APPROVED** [review]
LQGTM ! Thanks for this valuable contribution, I didn't had the time to check the whole PR in detail because they are so many changes.

**Comment by kmarchais** [comment]
Thank you both for the help!

**Pull request merged**